subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/InitialPassStatementTransformer.java

```diff
@@ -97,6 +99,21 @@ public Statement transform(SourceUnit sourceUnit, Statement statement) {
                 }
                 seenPluginRepositoriesBlock = true;
                 return statement; // don't transform the pluginRepositories block
+            } else if (scriptBlock.getName().equals(PLUGIN_MANAGEMENT)) {
```

Comment:
I'm not sure should we bother with this extraction at compile time. The script can simply use `Settings.pluginManagement()` at execution time. Same for the `pluginRepositories { }` statement.